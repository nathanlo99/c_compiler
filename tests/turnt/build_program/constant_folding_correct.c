int wain(int a, int b) {
  println(241);
  println((b - a));
  println(((a + b) + 2));
  return 1;
}
