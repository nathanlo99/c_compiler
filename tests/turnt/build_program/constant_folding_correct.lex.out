INT (int) at [1:1 - 1:3]
WAIN (wain) at [1:5 - 1:8]
LPAREN (() at [1:9 - 1:9]
INT (int) at [1:10 - 1:12]
ID (a) at [1:14 - 1:14]
COMMA (,) at [1:15 - 1:15]
INT (int) at [1:17 - 1:19]
ID (b) at [1:21 - 1:21]
RPAREN ()) at [1:22 - 1:22]
LBRACE ({) at [1:24 - 1:24]
PRINTLN (println) at [2:3 - 2:9]
LPAREN (() at [2:10 - 2:10]
NUM (241) at [2:11 - 2:13]
RPAREN ()) at [2:14 - 2:14]
SEMI (;) at [2:15 - 2:15]
PRINTLN (println) at [3:3 - 3:9]
LPAREN (() at [3:10 - 3:10]
LPAREN (() at [3:11 - 3:11]
ID (b) at [3:12 - 3:12]
MINUS (-) at [3:14 - 3:14]
ID (a) at [3:16 - 3:16]
RPAREN ()) at [3:17 - 3:17]
RPAREN ()) at [3:18 - 3:18]
SEMI (;) at [3:19 - 3:19]
PRINTLN (println) at [4:3 - 4:9]
LPAREN (() at [4:10 - 4:10]
LPAREN (() at [4:11 - 4:11]
LPAREN (() at [4:12 - 4:12]
ID (a) at [4:13 - 4:13]
PLUS (+) at [4:15 - 4:15]
ID (b) at [4:17 - 4:17]
RPAREN ()) at [4:18 - 4:18]
PLUS (+) at [4:20 - 4:20]
NUM (2) at [4:22 - 4:22]
RPAREN ()) at [4:23 - 4:23]
RPAREN ()) at [4:24 - 4:24]
SEMI (;) at [4:25 - 4:25]
RETURN (return) at [5:3 - 5:8]
NUM (1) at [5:10 - 5:10]
SEMI (;) at [5:11 - 5:11]
RBRACE (}) at [6:1 - 6:1]
